daphne_worker/src/config.rs

```diff
@@ -645,14 +647,29 @@ impl<'srv> DaphneWorker<'srv> {
     /// `hpke_config_id` is cached (if it exists).
     pub(crate) async fn get_hpke_receiver_config(
         &self,
-        hpke_receiver_kv_key: HpkeReceiverKvKey,
-    ) -> Result<Option<HpkeReceiverConfigKvPair>> {
-        self.kv_get_cached(
-            &self.isolate_state().hpke_receiver_configs,
-            KV_KEY_PREFIX_HPKE_RECEIVER_CONFIG,
-            Cow::Owned(hpke_receiver_kv_key),
-        )
-        .await
+        version: DapVersion,
+        config_id: u8,
+    ) -> Result<Option<HpkeReceiverConfig>> {
+        Ok(self
+            .get_hpke_receiver_configs(version)
+            .await?
+            .and_then(|mut configs| configs.remove(&config_id)))
```

Comment:
Why remove it? We're likely going to use the same `HpkeReceiverConfig` multiple times in the same request. If I understand the code correctly, we would end up hitting KV multiple times as a result.

Reply:
the method returns an Owned value, so any mutations on it doesn't affect anyone else. I use `remove` so that I can move the config out of the hashmap